components/backup_text/src/mask.rs

```diff
@@ -84,16 +85,19 @@ fn mask_string(bytes: &[u8]) -> Vec<u8> {
     hex.into_bytes()
 }
 
-fn mask_duration(dur: Duration) -> Duration {
-    let nanos = mask_i64(dur.to_nanos());
-    Duration::new(nanos, dur.fsp())
+fn mask_duration(dur: Duration) -> Result<Duration> {
+    // hack: 3e15 is slightly smaller than the max duration (838:59:59) * 10^9 nanosecs
+    let nanos = mask_i64(dur.to_nanos()) % 3_000_000_000_000_000;
+    let dur = Duration::new(nanos, 6).round_frac(dur.fsp() as i8)?;
+    Ok(dur)
 }
 
-fn mask_time(ctx: &mut EvalContext, time: DateTime) -> DateTime {
+fn mask_time(ctx: &mut EvalContext, time: DateTime) -> Result<DateTime> {
     let (fsp, time_type) = (time.fsp(), time.get_time_type());
 
     // Get a masked but unchecked time
-    let unchecked_time = DateTime(mask_u64(time.0));
+    let core_time = time.0 & (!0b1111);
```

Comment:
The least significant 4 bits of `DateTime` represent `fsp`, while the others represent `core_time` which keeps consistency with the TiDB side.

```go
const (
	// Core time bit fields.
	// ...

	// fspTt bit field.
	// `fspTt` format:
	// | fsp: 3 bits | type: 1 bit |
	// When `fsp` is valid (in range [0, 6]):
	// 1. `type` bit 0 represent `DateTime`
	// 2. `type` bit 1 represent `Timestamp`
	//
	// Since s`Date` does not require `fsp`, we could use `fspTt` == 0b1110 to represent it.
	fspTtBitFieldOffset, fspTtBitFieldWidth uint64 = 0, 4

	// ...
	fspTtBitFieldMask       uint64 = ((1 << fspTtBitFieldWidth) - 1) << fspTtBitFieldOffset

	coreTimeBitFieldMask        = ^fspTtBitFieldMask
)

// CoreTime returns core time.
func (t Time) CoreTime() CoreTime {
	return CoreTime(uint64(t.coreTime) & coreTimeBitFieldMask)
}
```

See also: https://github.com/BugenZhao/sql-masker/blob/d953e85be2/mask/funcs/workload_sim.go#L221-L222

Reply:
Plz add some comment